lib/msf/ui/console/command_dispatcher/jobs.rb

```diff
@@ -146,8 +146,12 @@ def cmd_jobs(*args)
                 job_id = val
               when "-p"
                 job_list = build_range_array(val)
-                job_list.each do |job_id|
-                  add_persist_job(job_id)
+                if !job_list.blank?
+                  job_list.each do |job_id|
+                    add_persist_job(job_id)
+                  end
+                else
+                  add_persist_job(nil)
```

Comment:
```suggestion
                job_list.each do |job_id|
                  add_persist_job(job_id)
                end
```

Just confirming; I thought we were updating the implementation to always return an array? 👀